sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/util/serializer/ObjectSerializer.java

```diff
@@ -18,36 +19,37 @@ public interface ObjectSerializer {
      * @param data The byte array.
      * @param typeReference {@link TypeReference} representing the object.
      * @param <T> Type of the object.
-     *
      * @return The object represented by the deserialized byte array.
+     * @throws IOException If the byte array cannot be deserialized.
      */
-    <T> T deserializeFromBytes(byte[] data, TypeReference<T> typeReference);
+    <T> T deserializeFromBytes(byte[] data, TypeReference<T> typeReference) throws IOException;
 
     /**
      * Reads a stream into its object representation.
      *
      * @param stream {@link InputStream} of data.
      * @param typeReference {@link TypeReference} representing the object.
      * @param <T> Type of the object.
-     *
      * @return The object represented by the deserialized stream.
+     * @throws IOException If the stream cannot be deserialized.
      */
-    <T> T deserializeFromStream(InputStream stream, TypeReference<T> typeReference);
+    <T> T deserializeFromStream(InputStream stream, TypeReference<T> typeReference) throws IOException;
```

Comment:
Should we add corresponding logs to implementing methods clarifying "Failed to deserialize object type"